pkg/kubectl/cmd/exec/exec.go

```diff
@@ -159,32 +181,39 @@ func (p *ExecOptions) Complete(f cmdutil.Factory, cmd *cobra.Command, argsIn []s
 		}
 		p.Command = argsIn
 	} else {
-		p.PodName = argsIn[0]
+		p.Resources = strings.Split(argsIn[0], "/")
```

Comment:
I don't think this is the right approach, the resource builder should already do the parsing for us. See how this is implemented in `kubectl logs`.

Reply:
@kargakis

Yeah, before i see how kubctl logs and kubectl port-forward implemented the resource builder.

I did this to addressing soltysh comment about passing the actual type and name.

I'm a little confused here :confused:

Reply:
This is not how `kubectl attach` (the example brought up by @soltysh ) parses arguments. In @soltysh's example the first argument in `ResourceNames` is the first argument passed in `kubectl attach` and the second argument in `ResourceNames` is the second arg in `kubectl attach` (see [here](https://github.com/kubernetes/kubernetes/blob/cee320a809e10d9147b634e4e2b05417c79a490b/pkg/kubectl/cmd/attach/attach.go#L106), then [here](https://github.com/kubernetes/kubernetes/blob/cee320a809e10d9147b634e4e2b05417c79a490b/pkg/kubectl/cmd/attach/attach.go#L179), then [here](https://github.com/kubernetes/kubernetes/blob/cee320a809e10d9147b634e4e2b05417c79a490b/pkg/kubectl/cmd/attach/attach.go#L226-L231)). What you have done here, is take the first argument in `kubectl exec`, split it, and pass those args in `ResourceNames`. This is not what @soltysh asked, but to his request I am not sure `kubectl exec` can support the `TYPE NAME` format because the second argument passed to `kubectl exec` is already parsed as a command to be executed in the pod (see [here](https://github.com/kubernetes/kubernetes/blob/cee320a809e10d9147b634e4e2b05417c79a490b/pkg/kubectl/cmd/exec/exec.go#L163)).

Reply:
@kargakis I see. 

I am not sure too what @soltysh is asking for is `kubectl exec` can support `TYPE NAME`. I have questioned this in my previous reply above. if what @soltysh requesting is `kubectl exec` can support `TYPE NAME`, I have no idea how implement this.

Reply:
@kargakis is right, I want to see these changes here: https://github.com/kubernetes/kubernetes/blob/cee320a809e10d9147b634e4e2b05417c79a490b/pkg/kubectl/cmd/attach/attach.go#L179 in `Complete` https://github.com/kubernetes/kubernetes/blob/cee320a809e10d9147b634e4e2b05417c79a490b/pkg/kubectl/cmd/attach/attach.go#L206-L214 in `Validate` and https://github.com/kubernetes/kubernetes/blob/cee320a809e10d9147b634e4e2b05417c79a490b/pkg/kubectl/cmd/attach/attach.go#L226-L231 in `Run`.

Reply:
@soltysh isn't the following suggestion incompatible with `kubectl exec` as it stands?
```go
 case 2: 
b.ResourceNames(o.Resources[0], o.Resources[1]) 
```
`kubectl exec` already parses the second argument (`o.Resources[1]`) as a command to be executed inside the pod (see [here](https://github.com/kubernetes/kubernetes/blob/cee320a809e10d9147b634e4e2b05417c79a490b/pkg/kubectl/cmd/exec/exec.go#L163))

Reply:
@soltysh can you point me about this one? 

how we can separate `TYPE` and `NAME` with whitespace if second argument in `kubectl exec` is a `COMMAND` that will be executed inside the pod? like my previous question how we can dealing with something like this?

| CMD          | ARG1      | ARG2    |
|--------------|-----------|---------|
| kubectl exec | POD       | COMMAND |
| kubectl exec | TYPE/NAME | COMMAND |
| kubectl exec | TYPE      | NAME    |

different with `kubectl attach` he don't have `COMMAND` as second argument. so it's fine `kubectl attach` meet something like this.

| CMD          | ARG1      | ARG2    |
|--------------|-----------|---------|
| kubectl attach | POD       | |
| kubectl attach | TYPE/NAME | |
| kubectl attach | TYPE      | NAME    |

like @kargakis say, i think my previous approach was quite right.

Reply:
@prksu sorry for going back and forth, please go ahead and revert to your previous PR state

Reply:
@kargakis i have reverted this to my previous approach.